agent-core/src/main/java/com/newrelic/agent/android/logging/RemoteLogger.java

```diff
@@ -0,0 +1,343 @@
+        }
+
+        Callable<Boolean> callable = () -> {
+            final Map<String, Object> logDataMap = new HashMap<>();
+
+            try {
+
+                /**
+                 * Some specific attributes have additional restrictions:
+                 *
+                 * accountId: This is a reserved attribute name. If it is included, it will be dropped during ingest.
+                 * appId: Must be an integer. When using a non-integer data type, the data will be ingested but becomes unqueryable.
+                 * entity.guid, entity.name, and entity.type: These attributes are used internally to identify entities.
+                 * Any values submitted with these keys in the attributes section of a metric data point may cause undefined behavior
+                 * such as missing entities in the UI or telemetry not associating with the expected entities.
+                 * eventType: This is a reserved attribute name. If it is included, it will be dropped during ingest.
+                 * timestamp: Must be a Unix epoch timestamp (either in seconds or in milliseconds) or an ISO8601-formatted timestamp.
+                 *
+                 * @link reserved attributes: https://source.datanerd.us/agents/agent-specs/blob/main/Application-Logging.md#log-record-attributes
+                 */
+                logDataMap.put(LogReporting.LOG_TIMESTAMP_ATTRIBUTE, String.valueOf(System.currentTimeMillis()));
+                logDataMap.put(LogReporting.LOG_LEVEL_ATTRIBUTE, logLevel.name().toUpperCase());
+
+                // set data with reserved attribute values
+                logDataMap.putAll(getCommonBlockAttributes());
+
+                // translate a passed message to attributes
+                if (message != null) {
+                    logDataMap.put(LogReporting.LOG_MESSAGE_ATTRIBUTE, message);
+                }
+
+                // translate any passed throwable to attributes
+                if (throwable != null) {
+                    logDataMap.put(LogReporting.LOG_ERROR_MESSAGE_ATTRIBUTE, throwable.toString());
+                    logDataMap.put(LogReporting.LOG_ERROR_STACK_ATTRIBUTE, throwable.getStackTrace()[0].toString());
+                    logDataMap.put(LogReporting.LOG_ERROR_CLASS_ATTRIBUTE, throwable.getClass().getSimpleName());
+                }
+
+                // finally add any passed attributes, which should not override reserved keys
+                if (attributes != null) {
+                    /**
+                     * "attributes" Object: This sub-object contains all other attributes of the message
+                     *
+                     * Number of attributes per event: 255 maximum.
+                     * Length of attribute name: 255 characters.
+                     * Length of attribute value: 4,094 characters are stored in NRDB as a Log event field
+                     **/
+
+                    logDataMap.put(LogReporting.LOG_ATTRIBUTES_ATTRIBUTE, attributes);
```

Comment:
why we are not iterating the attributes instead of appending with our log default attribute?

Reply:
There aere two Json models supprted by Log ingest: flat (with all attributes on teh root), and common block (with attributes under their own block). I'd like to see which works better for us. Again, to be resolved by GA.

Reply:
I was also curious if we might want to isolate customer provided attributes from those we decorate the log message .